### ISSUE TITLE
Page load performance improvements on blogfarm

### ISSUE THREAD

**Issue by rufuspollock** [okfn/okfn.org#83]
(was:Think about our page sizes (discussion))

First, what are our options:
- Reduce page sizes (see below) 
  - Be more careful about using images
- Use a CDN more - #91 
  - one attractive aspect would be that we could centralize on WP again as our host point for image files ...
- Reduce the number of different things we are loading e.g.
  - Do we need tweetblender loading on every site and every page
  - Ditto for share this buttons - how do you switch them off ??
## Page Sizes

Currently at 1.8MB for okfn.org an 1Mb for blog.okfn.org. Page load times > 1s and substantially more for actual users (image loading etc)

See e.g. 

http://tools.pingdom.com/fpt/#!/gnAOVje21/okfn.org

http://tools.pingdom.com/fpt/#!/t1bpd6AjR/http://blog.okfn.org

**Comment by smth** [comment]
Seems to me that all the large chunks of time are coming from externally hosted files (which is probably to be expected), so one obvious solution would be host everything ourselves.. On one level this could be not using remotely hosted js files, on another level this could mean being more strict with our content. We shouldn't be hot linking files like this for example http://epsiplatform.eu/sites/default/files/warsawskyline.png (copyright issues aside).

**Comment by rdonovan** [comment]
Externally hosted js/css etc is typically a good thing as a dedicated CDN is going to be much faster at serving these types of files than our server. But I suspect @mintcanary you're rather referring to sporadic images here. 

The browser extension Yslow gives a really good insight into the problem areas of performance for websites. For okfn.org this appears to highlight better use of CDN's and asset minification/compression as our main areas for improvement. 

I've employed the WP Minify plugin with a great deal of success on other sites to dynamically address combining and minifying assets to reduce page size by a significant amount previously. Perhaps worth a shot here also? Keep in mind this will introduce yet another level of caching into the workflow. And I know how much @mintcanary loves dealing with the existing cache quirks.

YSlow
Chrome - https://chrome.google.com/webstore/detail/yslow/ninejjcohidippngpapiilnmkgllmakh?hl=en
Firefox - https://addons.mozilla.org/en-us/firefox/addon/yslow/

WP Minify
http://wordpress.org/extend/plugins/wp-minify/

**Comment by smth** [comment]
Yes, most of the JS will be fine, and the images do seem to be mostly to blame. In terms of JS, I was thinking of the odd service which seems to noticeably slow down page loading. I often notice (on various sites) that 'share' buttons take forever to load, for example. ShareThis is one plugin we could perhaps ditch, in favour of integrating something like this http://filamentgroup.com/lab/socialcount

**Comment by rufuspollock** [comment]
@mintcanary @rdonovan - just refactored title and description to focus on really issues.

Aside: +1 one on ditching sharethis for something more customizable and flexible ...

**Comment by smth** [comment]
ShareThis: #90

**Comment by smth** [comment]
So nobody repeats the process that I just went through, we are not ditching ShareThis as a service.
